### DISCUSSION TITLE
Implementing Collission Event or Functions

### DISCUSSION THREAD

**Discussion by Hazeofdream** [wiremod/wire#2733] [Suggestions]
Exist in the form of https://steamcommunity.com/sharedfiles/filedetails/?id=689046752

As a decreased reliance on cores becomes common, One of the most common uses and most frustrating things when it comes to Expression2 is collision detection, usually when it comes to missiles or other weaponry.

E2 has this amazing ability to either have really janky collisions that don't work half the time or are so intensive that it will crash your e2 in no time.

In this case, Why not transfer CollisionCore into an event system? or functions alternatively (or both)

**Comment by Hazeofdream** [comment]
@Vurv78 because he is the event godfather and probably finds this easier then me to implement

**Comment by vurvdev** [comment]
There's already an addon that gives you this functionality, it won't be added to base wire for the same reason why there's no timer event, it doesn't really make sense to handle something that works per-instance as a global `event`.

In the future it'd be like starfall with function objects:

```
e:onCollide(function() {
 # do stuff
})
```
